roles/matrix-bot-maubot/templates/systemd/matrix-maubot.service.j2

```diff
@@ -0,0 +1,40 @@
+#jinja2: lstrip_blocks: "True"
+[Unit]
+Description=Maubot
+{% for service in matrix_bot_maubot_systemd_required_services_list %}
+Requires={{ service }}
+After={{ service }}
+{% endfor %}
+{% for service in matrix_bot_maubot_systemd_wanted_services_list %}
+Wants={{ service }}
+{% endfor %}
+DefaultDependencies=no
+
+[Service]
+Type=simple
+Environment="HOME={{ matrix_systemd_unit_home_path }}"
+ExecStartPre=-{{ matrix_host_command_sh }} -c '{{ matrix_host_command_docker }} kill matrix-maubot 2>/dev/null || true'
+ExecStartPre=-{{ matrix_host_command_sh }} -c '{{ matrix_host_command_docker }} rm matrix-maubot 2>/dev/null || true'
+
+ExecStart={{ matrix_host_command_docker }} run --rm --name matrix-maubot \
+			--log-driver=none \
```

Comment:
Could you test starting the container directly with the matrix user and group, like we do for our other services?

`--user={{ matrix_user_uid }}:{{ matrix_user_gid }}`, instead of starting as root and expecting that the container will drop to whatever specify in the `UID` and `GID` environment variables.

This may mean that we won't use its own entrypoint.. And that likely we won't need to be passing `UID`/`GID` anymore.

Along the same lines, would this work if we add `--cap-drop=ALL`, so that we can further reduce the container process' priviledges?

Can we also use `--read-only`? For this, you may need to first adjust the bot's logging configuration, so that it won't try to write to `/var/log`.

Reply:
--user is not possible as it fails with 
```
Jun 24 15:05:08 X-Wing matrix-maubot[1016487]: mkdir: can't create directory '/var/log/maubot': Permission denied
Jun 24 15:05:09 X-Wing matrix-maubot[1016487]: FAILED: No config file 'alembic.ini' found, or file has no '[alembic]' section
Jun 24 15:05:09 X-Wing matrix-maubot[1016487]: chown: /var/log: Operation not permitted
Jun 24 15:05:09 X-Wing matrix-maubot[1016487]: chown: /var/log: Operation not permitted
Jun 24 15:05:09 X-Wing matrix-maubot[1016487]: su-exec: setgroups(1000): Operation not permitted

```

Reply:
Also `--cap-drop=ALL`
```

Jun 24 15:06:42 X-Wing matrix-maubot[1017218]: Error: open /data/config.yaml: permission denied
Jun 24 15:06:42 X-Wing matrix-maubot[1017218]: Error: open /data/config.yaml: permission denied
Jun 24 15:06:42 X-Wing matrix-maubot[1017218]: su-exec: setgroups(1000): Operation not permitted

```

Reply:
Read-only is done

Reply:
Looking at the entrypoint command [here](https://github.com/maubot/maubot/blob/47d7e2b483aa598a601231b88bbd4af78f71e4c6/docker/run.sh), it seems like this script does a initialization stuff like this:
- creating some directories under `/data` (`/data/plugins`, `/data/trash`, `/data/dbs`) - we may be able to pre-create these via the playbook (just add them to `setup_install.yml`)
- creates the `/var/log` directory in the container. We'd ideally stop logging to files, as discussed elsewhere. It's probably done the same way it's done for other mautrix bridges. See [our configuration for mautrix-facebook](https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/bd05e0c1b46bfb0acc0e540535a5efafb12c41dc/roles/matrix-bridge-mautrix-facebook/templates/config.yaml.j2#L235-L260). If we can't stop it, we can probably mount a `tmpfs` volume at `/var/log`.
- updates the configuration (relative to absolute paths) - we can skip this by defining absolute paths in our configuration from the beginning, instead of relying on the script to fix it up. This is probably already the case, so maybe there is nothing to do about it.
- moves some databases that were previously in `/data/plugins` to `/data/dbs` - this seems like a data migration from some olden times where databases were stored in `/data/plugins`. We don't need to care about it

With these changes, we may be able to start with `--user` and `--cap-drop=ALL`, and to invoke `python3 -m maubot -c /data/config.yaml` directly, instead of using [the default command specified in the Dockerfile](https://github.com/maubot/maubot/blob/47d7e2b483aa598a601231b88bbd4af78f71e4c6/Dockerfile#L63).

To start the continer with a custom command, simply append it at the end of the `docker run` invocation in this `.service` file - [we do a similar thing for Synapse](https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/bd05e0c1b46bfb0acc0e540535a5efafb12c41dc/roles/matrix-synapse/templates/synapse/systemd/matrix-synapse.service.j2#L64).

Reply:
done